common/src/main/java/com/wynntils/services/itemfilter/SearchQuery.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import com.wynntils.core.components.Services;
+import com.wynntils.models.items.WynnItem;
+import com.wynntils.utils.type.ErrorOr;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.IntStream;
+
+public class SearchQuery {
+    private final String queryString;
+
+    private final List<ItemFilter> itemFilters;
+
+    private final List<String> plainTextTokens;
+
+    private final List<Integer> ignoredCharIndices;
+
+    private final List<Integer> validFilterCharIndices;
+
+    private final List<String> errors;
```

Comment:
You can remove that spacing, we only put them in case there is a reasonable logic separation. This is all the same logic.